lib/cylc/tests/test_task_id.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python2
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from cylc.task_id import TaskID
+
+
+class TestTaskId(unittest.TestCase):
+
+    def test_get(self):
+        self.assertEqual("a.1", TaskID.get("a", 1))
+        self.assertEqual("a._1", TaskID.get("a", "_1"))
+        self.assertEqual(
+            "WTASK.20101010T101010", TaskID.get("WTASK", "20101010T101010"))
+
+    def test_split(self):
+        self.assertEqual(["a", '1'], TaskID.split("a.1"))
+        self.assertEqual(["a", '_1'], TaskID.split("a._1"))
+        self.assertEqual(
+            ["WTAS", '20101010T101010'], TaskID.split("WTAS.20101010T101010"))
+
+    def test_is_valid_name(self):
+        for name in [
+            "abc", "123", "____", "_", "a_b", "a_1", "1_b", "ABC"
+        ]:
+            self.assertTrue(TaskID.is_valid_name(name))
+        for name in [
+            "a.1", None, "%abc", "", " "
```

Comment:
Perhaps add: `a b`, `a\nb`

Reply:
Looking forward to closing #2281 when we move to Python 3!

Reply:
When, hopefully `\w+` will be sufficient.